src/core/dbus-execute.c

```diff
@@ -1701,22 +1709,28 @@ int bus_exec_context_set_transient_property(
 
                 return 1;
 
-        } else if (streq(name, "RuntimeDirectoryMode")) {
+        } else if (STR_IN_SET(name, "RuntimeDirectoryMode", "StateDirectoryMode", "CacheDirectoryMode", "LogsDirectoryMode", "ConfigurationDirectoryMode")) {
                 mode_t m;
 
                 r = sd_bus_message_read(message, "u", &m);
                 if (r < 0)
                         return r;
 
                 if (mode != UNIT_CHECK) {
-                        c->runtime_directory_mode = m;
+                        int i;
```

Comment:
not that it matters much, but I think using "ExecDirectoryType" instead of "int" as type of `i` would make this prettier... After all we iterate through all known types here...